lib/rules/no-unsupported-role-attributes.js

```diff
@@ -0,0 +1,84 @@
+import { roles, elementRoles } from 'aria-query';
+
+import AstNodeInfo from '../helpers/ast-node-info.js';
+import Rule from './_base.js';
+
+function createUnsupportedAttributeErrorMessage(attr, role, element) {
+  if (element) {
+    return `The attribute ${attr} is not supported by the element ${element} with the implicit role of ${role}`;
+  } else {
+    return `The attribute ${attr} is not supported by the role ${role}`;
+  }
+}
+function getImplicitRole(element, typeAttribute) {
+  if (element === 'input') {
+    for (let key of elementRoles.keys()) {
+      if (key.name === element) {
+        let attributes = key.attributes;
+        for (let attribute of attributes) {
+          if (attribute.name === 'type' && attribute.value === typeAttribute) {
+            return elementRoles.get(key)[0];
+          }
+        }
+      }
+    }
+  } else {
+    return elementRoles.get(elementRoles.keys().find((key) => key.name === element))[0];
+  }
+}
+
+export default class NoUnsupportedRoleAttributes extends Rule {
+  visitor() {
+    return {
+      ElementNode(node) {
+        let element, typeAttribute, role;
+        if (AstNodeInfo.hasAttribute(node, 'role')) {
+          let roleAttrNode = AstNodeInfo.findAttribute(node, 'role');
+          if (roleAttrNode.value.type === 'TextNode') {
+            role = roleAttrNode.value.chars || '';
```

Comment:
I think `null` or `undefined` is more clear for indicating that we couldn't find a role / type.